### PR TITLE
config: add config metrics

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

Add following config metric, this is for DBaaS alter management.
* `token-limit`
* `mem-quota-query`
* `max-server-connections`

### What is changed and how it works?

N/A

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

image

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM

### Release note 

- N/A

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -309,6 +309,9 @@ func setTxnScope() {
 // Export config-related metrics
 func (s *Server) reportConfig() {
 	metrics.TokenLimitGauge.Set(float64(s.cfg.TokenLimit))
```
Comment (crazycs520) [inline server/server.go:311]: How about remove the `TokenLimitGauge ` metric and use `ConfigStatus` with the `type="token-limit"`?  @iosmanthus

**Review by breeswish — APPROVED** [review]

File: server/server.go (hunk 1 above)
Comment (iosmanthus) [reply server/server.go:311]: Cool

Comment (iosmanthus) [reply server/server.go:311]: I will create another pull request and cherry-pick this change.

**Pull request merged**